src/irmin-mirage/git/irmin_mirage_git.mli

```diff
@@ -0,0 +1,164 @@
+module type S = sig
+  include Irmin_git.S with type Private.Sync.endpoint = Git_mirage.endpoint
+
+  val remote :
+    ?conduit:Conduit_mirage.conduit ->
+    ?resolver:Resolver_lwt.t ->
+    ?headers:Cohttp.Header.t ->
+    string ->
+    Irmin.remote
+end
+
+module Git : sig
+  module type S_MAKER = functor
+    (G : Irmin_git.G)
+    (C : Irmin.Contents.S)
+    (P : Irmin.Path.S)
+    (B : Irmin.Branch.S)
+    -> S
+       with type key = P.t
+        and type step = P.step
+        and module Key = P
+        and type contents = C.t
+        and type branch = B.t
+        and module Git = G
+
+  module type KV_MAKER = functor (G : Irmin_git.G) (C : Irmin.Contents.S) -> S
```

Comment:
ouch, the formatting here is just terrible. Is this via ocamlformat? This deserves a bug report if so

Reply:
Yeah, it seems to be from ocamlformar, I will open an issue about this.

Reply:
Actually, there appears to be an issue about this already: https://github.com/ocaml-ppx/ocamlformat/issues/671 and it looks like it's been fixed. I will investigate a little more.